js/src/Ice/InstanceExtensions.js

```diff
@@ -314,15 +309,21 @@ Instance.prototype.finishSetup = function (communicator) {
         }
     } catch (ex) {
         if (ex instanceof LocalException) {
-            this.destroy();
+            this.destroy().catch(err => {
```

Comment:
The destroy operation is "async", without this catch we can end-up with an unhandled promise rejection.